client/client.go

```diff
@@ -149,6 +142,14 @@ func (c *Client) createRequest(method, url string, bodyObject interface{}) (req
 	req.Header.Set("Connection", "close")
 	req.Close = true
 
+	// add any passed in headers
+	if headers != nil {
+		for k, v := range headers {
+			c.logPrintf("Client.createRequest() setting header: key=%q, value=%q", k, v)
```

Comment:
might want to relocate the `c.logPrintf` to the `defer func` above, looks like logging is only done if the logger isn't `nil`

Reply:
Done, but `logPrintf` also checks if logger is nil.